packages/next/src/lib/metadata/get-metadata-route.ts

```diff
@@ -11,14 +11,33 @@ import { normalizePathSep } from '../../shared/lib/page-path/normalize-path-sep'
  * Give it a unique hash suffix to avoid conflicts
  *
  * e.g.
- * /app/opengraph-image.tsx -> /opengraph-image
- * /app/(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ * /opengraph-image -> /opengraph-image
+ * /(post)/opengraph-image.tsx -> /opengraph-image-[0-9a-z]{6}
+ *
+ * Sitemap is an exception, it should not have a suffix.
+ * As the generated urls are for indexer and usually one sitemap contains all the urls of the sub routes.
```

Comment:
I'm having a hard time following this sentence -- I think we can just clarify that the sitemap has to be accessible at a predictable URL so that crawlers can find it.

Reply:
Updated the comments a bit